simple/stats/nl.py

```diff
@@ -53,9 +65,58 @@ def generate_nl_sentences(triples: list[Triple], sentences_fh: FileHandler):
 
   dataframe = pd.DataFrame(rows)
 
+  sentences_fh = nl_dir_fh.make_file(_SENTENCES_FILE)
   logging.info("Writing %s NL sentences to: %s", dataframe.size, sentences_fh)
   sentences_fh.write_string(dataframe.to_csv(index=False))
 
+  embeddings_dir_fh = nl_dir_fh.make_file(f"{_EMBEDDINGS_DIR}/")
+  embeddings_dir_fh.make_dirs()
+  embeddings_fh = embeddings_dir_fh.make_file(_EMBEDDINGS_FILE)
+  catalog_fh = embeddings_dir_fh.make_file(_CUSTOM_CATALOG_YAML)
+  catalog_dict = _catalog_dict(nl_dir_fh.path, embeddings_fh.path)
+  catalog_yaml = yaml.safe_dump(catalog_dict)
+  logging.info("Writing custom catalog to path %s:\n%s", catalog_fh,
+               catalog_yaml)
+  catalog_fh.write_string(catalog_yaml)
+
+
+def _catalog_dict(nl_dir: str, embeddings_path: str) -> dict:
+  return {
+      "version": "1",
+      "indexes": {
+          _CUSTOM_EMBEDDING_INDEX: {
+              "store_type": "MEMORY",
+              "source_path": nl_dir,
+              "embeddings_path": embeddings_path,
+              "model": _CUSTOM_MODEL
+          },
+      },
+      "models": {
+          _CUSTOM_MODEL: {
+              "type": "LOCAL",
+              "usage": "EMBEDDINGS",
+              "gcs_folder": _CUSTOM_MODEL_PATH,
+              "score_threshold": 0.5
+          }
+      }
+  }
+
+
+def _rewrite_catalog_for_testing(catalog_fh: FileHandler):
```

Comment:
Don't like this approach but can't think of an alternative. Let me know if any suggestions.

Reply:
Left a suggestion

Reply:
Thanks.